File: ZendAMF/app/servicos/ServicosPreVenda.php
<?php
//include '/home2/pcmaisma/public_html/admProjetos/ZendAMFAdminProjeto/app/db/BaseDados.php';
//include '/home2/pcmaisma/public_html/admProjetos/ZendAMFAdminProjeto/app/vo/ProjetoVO.php';

//include 'app/db/BaseDados.php';
include 'app/vo/PreVendaVO.php';
include 'app/vo/ItemPrevendaVO.php';

class ServicosPreVenda {
	
	
	private $conn;
	
	
	function ServicosPreVenda() {
		$db = new BaseDados();
		$this->conn = $db->conn;
	}
	
	
	public function abrirPreVenda(PreVendaVO $item){
		$sql = "insert into prevenda (codUsuario,status,dataAbertura) values ('$item->codUsuario'		 
		 ,'0' ,now())";
		$this->conn->StartTrans();
		$resultado = $this->conn->Execute($sql);
		$item->codigo = $this->conn->insert_Id();
		
		if($this->conn->HasFailedTrans()){
			throw new Exception("Erro ao abri pré-Venda",16);
		}	
		
		$this->conn->CompleteTrans();
		$item->status = 0;
		
		return $item;
	}
	
	public function addItemPreVenda(ItemPreVendaVO $item){
			
			
			
			$sql ="select * from produto where codigo = $item->codProduto";
			
			$this->conn->StartTrans();
			
			$result = $this->conn->Execute($sql);	
			
			
			if(!$result){
				throw new Exception("Produto não existe",4);
			}
			$registro = $result->FetchNextObject();
			
			
			if($registro->QTDEMESTOQUE < $item->quantidade){
				throw new Exception("Quantidade de produtos maior que disponível!",15);
			}
			
			
			$sql = "insert into itensprevenda (codPrevenda, codProduto, quantidade,valor) 
			      values ('$item->codigoPrevenda','$item->codProduto','$item->quantidade','$item->valor')";
			$this->conn->Execute($sql);
			
					
			$sql = "UPDATE produto SET qtdEmEstoqu = (qtdEmEstoqu - $item->quantidade) where codigo = $item->codProduto";
						
			$result = $this->conn->Execute($sq);
			
			$falhou = false;	
			if($this->conn->HasFailedTrans()){
				$falhou = true;
			}
			
			$this->conn->CompleteTrans();
			
			if($falhou){
				throw new Exception("Erro ao inserir item!",16);
			}
			
			
			return $item;
			
	}
	
	
//	foreach ($item->itemPreVenda as $temp) {
//			$sql = "insert into itensprevenda (codPrevenda, codProduto, quantidade,valor) 
//			      values ('$item->codigo','$temp->codProduto','$temp->quantidade','$temp->valor')";
//			$this->conn->Execute($sql);
//		}
	
	public function fecharPreVenda(PreVendaVO $item){
		if($item->codCliente==0){
			$sql = "UPDATE prevenda SET codUsuario = '$item->codUsuario' , status = '1', obs = '$item->obs' where codigo = $item->codigo";
		}else{
			$sql = "UPDATE prevenda SET codUsuario = '$item->codUsuario' , codCliente = '$item->codCliente', status = '1', obs = '$item->obs' where codigo = $item->codigo";	
		}
		$this->conn->Execute($sql);
		return $item;
	}
	
	public function cancelarPreVenda(PreVendaVO $item){
		$sql = "UPDATE prevenda SET codUsuario = '$item->codUsuario' , status = '2', obs = '$item->obs' where codigo = $item->codigo";
		$this->conn->Execute($sql);
		return $item;
	}
	
	
	public function removerItem(PreVendaVO $item){
		
		$sql = "delete from prevenda  where codigo = $item->codigo";
		$resultado = $this->conn->Execute($sql);
		if($this->conn->Affected_Rows()==0){
			return false;
		}
		return true;
	}
	
	public function pesquisarItens($texto,$coluna){
		$sql = "select * from prevenda where $coluna like '%$texto%'";
		$resultado = $this->conn->Execute($sql);
		while($registro = $resultado->FetchNextObject()){			
			$dados_item = new PreVendaVO();
			$dados_item->codigo = $registro->CODIGO;
			$dados_item->codCliente = $registro->CODCLIENTE;
			$dados_item->codUsuario = $registro->CODUSUARIO;
			$dados_item->obs = $registro->OBS;
			$dados_item->status = $registro->STATUS;
			$dados_item->dataAbertura = $registro->DATAABERTURA;
			$retorna_dados_item [] = $dados_item;
		}
		return $retorna_dados_item;
	}
	
	public function getItens(){
		$sql = "select * from projeto";
		$resultado = $this->conn->Execute($sql);
		while($registro = $resultado->FetchNextObject()){			
			$dados_item = new PreVendaVO();
			$dados_item->codigo = $registro->CODIGO;
			$dados_item->codCliente = $registro->CODCLIENTE;
			$dados_item->codUsuario = $registro->CODUSUARIO;
			$dados_item->obs = $registro->OBS;
			$dados_item->status = $registro->STATUS;
			$dados_item->dataAbertura = $registro->DATAABERTURA;
			$retorna_dados_item [] = $dados_item;
		}
		return $retorna_dados_item;	
	}
}
//$eu = new ServicosPreVenda();
////$c = new PreVendaVo();
////$c->codUsuario = 0;
//$eu->addItemPreVenda(new ItemPreVendaVO());

?>

File: ServidorJava/WEB-INF/src/services/ServicosPreVenda.java
import java.sql.ResultSet;
import java.util.ArrayList;

import vo.ItemPreVendaVO;
import vo.PreVendaVO;
import arquitetura.BancoDeDados;


public class ServicosPreVenda {
	
	

	
	

	
	
	public PreVendaVO abrirPreVenda(PreVendaVO item){
	//	f = fopen('log2.txt',"w+");
	String sql = "insert into prevenda (codUsuario,status,dataAbertura) values ("+item.codUsuario+",0 ,now())";
		
		
		if(banco.executarNoQuery(sql)==0){
			throw new RuntimeException("Erro ao Adicionar Cliente");
		}	
		
		
		return item;
	}
	
	public ItemPreVendaVO addItemPreVenda(ItemPreVendaVO item){
		String	sql ="select * from produto where codigo = "+item.codProduto;
			
		ResultSet result = banco.executar(sql);	
		ArrayList<ItemPreVendaVO> itens  = this.toItemPreVenda(result);
		if(itens.size()==0){
				throw new RuntimeException("Produto não existe");
			}
			ItemPreVendaVO registro = itens.get(0);
			if(registro.quantidade < item.quantidade){
				throw new RuntimeException("Quantidade de produtos maior que disponível!");
			}
			sql = "insert into itensprevenda (codPrevenda, codProduto, descricao, quantidade,valor)"+ 
			      "values ('"+item.codigoPrevenda+"','"+item.codProduto+"','"+item.descricao+"','"+item.quantidade+"','"+item.valor+"')";
			if(banco.executarNoQuery(sql)==0){
				throw new RuntimeException("Erro ao Adicionar ItemPrevenda");
			}
					
			sql = "UPDATE produto SET qtdEmEstoque = (qtdEmEstoque - "+item.quantidade+") where codigo = "+item.codProduto;
			
			if(banco.executarNoQuery(sql)==0){
				throw new RuntimeException("Erro ao atualizar itemprevenda");
			}
			
			
			return item;
			
	}
	
	
//	foreach (item.itemPreVenda as temp) {
//			sql = "insert into itensprevenda (codPrevenda, codProduto, quantidade,valor) 
//			      values ('item.codigo','temp.codProduto','temp.quantidade','temp.valor')";
//			this.conn.Execute(sql);
//		}
	
	public PreVendaVO fecharPreVenda(PreVendaVO item){
String sql = "UPDATE prevenda SET codUsuario = '"+item.codUsuario+"' , status = '1', obs = '"+item.obs+"', valorTotal = '"+item.valorTotal+"' where codigo = "+item.codigo+"";		
if(banco.executarNoQuery(sql)==0){
	throw new RuntimeException("Erro ao fechar Prevenda");
}
		return item;
	}
	
	public void removerItemPreVenda(ItemPreVendaVO item){
		
		String sql = "UPDATE produto SET qtdEmEstoque = (qtdEmEstoque + "+item.quantidade+")  WHERE codigo = "+item.codProduto;
		if(banco.executarNoQuery(sql)==0){
			throw new RuntimeException("Erro ao atualizar ItemPrevenda");
		}
		sql = "DELETE FROM itensprevenda WHERE codigo = "+item.codigo;
		if(banco.executarNoQuery(sql)==0){
			throw new RuntimeException("Erro ao Deletar ItemPrevenda");
		}
	}
	
	public PreVendaVO cancelarPreVenda(PreVendaVO item){	
		String sql = "select * FROM itensprevenda WHERE codPrevenda = item.codigo";
		this.conn.StartTrans();
		ResultSet resultado = banco.executar(sql);
		ArrayList<ItemPreVendaVO> itens= this.toItemPreVenda(resultado);
		for (int i = 0; i < itens.size(); i++) {
			ItemPreVendaVO registro;
		
			sql = "UPDATE produto SET qtdEmEstoque = (qtdEmEstoque + "+registro.quantidade+")  WHERE codigo = "+registro.codProduto;
			if(banco.executarNoQuery(sql)==0){
				throw new RuntimeException("Erro ao atualizar produto");
			}		
		}
		sql = "DELETE FROM itensprevenda WHERE codPrevenda = "+item.codigo;
		if(banco.executarNoQuery(sql)==0){
			throw new RuntimeException("Erro ao Deletar Prevenda");
		}
		sql = "UPDATE prevenda SET codUsuario = '"+item.codUsuario+"' , status = '2', obs = '"+item.obs+"' where codigo = "+item.codigo;
		if(banco.executarNoQuery(sql)==0){
			throw new RuntimeException("Erro ao Atualizar Prevenda");
		}
		return item;
	}
	
	
	public boolean removerItem(PreVendaVO item){
		
		String sql = "delete from prevenda  where codigo = item.codigo";
		
		if(banco.executarNoQuery(sql)==0){
			return false;
		}else{
			return true;
		}
	}
	
	public ArrayList<PreVendaVO> filtraData(String data,String coluna){
		String sql = "select * from prevenda where DATE("+coluna+") = '"+data+"' and status = 1";
		ResultSet result = banco.executar(sql);
		return this.toPreVenda(result);
	}
	
	private ArrayList<PreVendaVO> getItensPrevenda(String codigo){
		String sql = "select * from itensprevenda where codPrevenda  = "+codigo;
		ResultSet resultado = banco.executa(sql);
		
		return this.toPreVenda(resultado);
	}
	
	
	
	public ArrayList<PreVendaVO> pesquisarItens(String texto,String coluna){
      String sql = "select * from prevenda where "+coluna+" like '%"+texto+"%'";
		ResultSet resultado = banco.executar(sql);
		
		return this.toPreVenda(resultado);
	}
	
	public ArrayList<PreVendaVO> getItens(){
		String sql = "select * from prevenda";
	
	ResultSet rs = banco.executar(sql);
		return this.toPreVenda(rs);	
	}
	
	public ArrayList<PreVendaVO> getItensValidos(){
		String sql = "select * from prevenda where status = 1";
		ResultSet rs = banco.executar(sql);
		return this.toPreVenda(rs);	
	}
	
	public ArrayList<PreVendaVO> pesquisarItensValidos(String texto,String coluna){
		String sql = "select * from prevenda where "+coluna+" like '%"+texto+"%' and status = 1";
		ResultSet rs = banco.executar(sql);
		return this.toPreVenda(rs);
	}
	
	
	private ArrayList<PreVendaVO> toPreVenda(ResultSet rs){
		  ArrayList<PreVendaVO> gp = new ArrayList<PreVendaVO>();
		while (rs.next()) {
			PreVendaVO dados_item = new PreVendaVO();
			dados_item.codigo = rs.getInt("codigo");
			dados_item.codUsuario = rs.getInt("codUsuario");
			dados_item.dataAbertura = rs.getDate("dataAbertura");
			dados_item.obs = rs.getString("obs");
			dados_item.status = rs.getInt("status");
			dados_item.valorTotal = rs.getInt("valorTotal");
			
			
			
			

			gp.add(dados_item);
		}
		return gp;
		}
	
	
	private ArrayList<ItemPreVendaVO> toItemPreVenda(ResultSet rs) {
		
	      ArrayList<ItemPreVendaVO> gp = new ArrayList<ItemPreVendaVO>();
	    while (rs.next()) {
	    	ItemPreVendaVO dados_item = new ItemPreVendaVO();
		dados_item.codigo = rs.getInt("codigo");
		dados_item.codigoPrevenda = rs.getInt("codPrevenda");
		dados_item.codProduto = rs.getInt("codProduto");
		dados_item.descricao = rs.getString("descricao");
		dados_item.quantidade = rs.getInt("quantidade");
		dados_item.valor = rs.getDouble("valor");
		
		

		gp.add(dados_item);
	}
	return gp;
	}
}
